### PR TITLE
Feature/process results

### PR DESCRIPTION
**Pull request by otfrom**

### CODE REVIEW

File: src/witan/workspace/coercion.clj [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    clojure.lang.PersistentArrayMap (map (partial replacer rfn) x)
+    java.lang.String (rfn x)))
+
+(defn underscore->hyphen
+  "Converts underscores to hyphens"
+  [x]
+  (replacer #(clojure.string/replace % #"_" "-") x))
+
+(defn hyphen->underscore
+  "Convers hyphens to underscores"
```
Comment (tcoupland) [inline src/witan/workspace/coercion.clj]: Spelling mistake

File: src/witan/workspace/command.clj [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+            [witan.workspace.schema     :as ws]
+            [witan.workspace.protocols  :as p]
+            [schema.core                :as s]
+            [schema-contrib.core        :as sc]
+            [schema.coerce              :as coerce]
+            [clojure.stacktrace         :as st]
+            [witan.gateway.schema       :as wgs]
+            [witan.workspace-api.schema :as was]
+            [witan.workspace.coercion   :as wc]
+            [witan.workspace.event      :as ev])
```
Comment (tcoupland) [inline src/witan/workspace/command.clj:13]: Very tidy, but not in alphabetical order

Comment (acron0) [inline src/witan/workspace/command.clj:13]: srsly?

File: src/witan/workspace/event.clj [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+  (log/info "Sending event:" (:event/key event) (:event/version event))
+  (p/send-message! sender :event event))
+
+(defn create-event
+  [receipt body]
+  (let [iso-date-time-now (->> (java.util.Date.)
+                               (json/generate-string)
+                               (drop 1)
+                               (butlast)
+                               (apply str))]
```
Comment (tcoupland) [inline src/witan/workspace/event.clj]: Would have expected to see a clojure.time formatter here...

Comment (otfrom) [inline src/witan/workspace/event.clj]: I missed that we'd already pulled in clj-time

File: src/witan/workspace/command.clj (hunk 2 above)
Comment (tcoupland) [inline src/witan/workspace/command.clj:13]: no. Of course not :)

File: src/witan/workspace/util.clj [hunk 4]
```diff
@@ -26,60 +15,10 @@
...
-        :else nil))))
-
-(defn param-coercion-matcher
-  [schema]
-  (or (coerce/json-coercion-matcher schema)
-      (workflow-node-coercion schema)
-      (uuid-coercion schema)))
+(defn timestamp
+  ([fmt time]
+   (tf/unparse (tf/formatters fmt) time))
```
Comment (tcoupland) [inline src/witan/workspace/util.clj]: There's a horrible outstanding java bug around synchronizing the fetching of time formatters. Wrap this in a little memoized function and things will flow a bit easier!

Comment (otfrom) [inline src/witan/workspace/util.clj]: @mantree do you have a link to the bug?

Comment (tcoupland) [inline src/witan/workspace/util.clj]: can't find it :( Might even be fixed now!

**Comment by tcoupland** [comment]
:shipit:

**Pull request merged**